### PR TITLE
Localize HeliosLauncher UI using lang files

### PR DESCRIPTION
**Pull request by Kamesuta**
## This pull request adds the following features:
- Externalize the UI text into lang files and replace them with embedded codes using LangLoader.
  - In ejs files, use the ejs output tag (`<%- lang('～') %>`) syntax.
- Support argument expansion in LangLoader. Expand the object in the second argument from `{placeHolder}`.
- Replace json with toml for easier element addition and indentation.
  - If you don't like toml, you can revert the last commit to go back to json.

### CODE REVIEW

**Comment by dscalzi** [comment]
I will take a look, this would be a very nice addition

**Comment by luisBazanDev** [comment]
It is a very good implementation, I hope you can accept it as soon as possible so I can do the translation into Spanish.

**Comment by Codixer** [comment]
I wonder the state of this 0_0

**Comment by Kamesuta** [comment]
## Updated: Added cascadable language system for easy customization by launcher forkers.
- Translators can now change the language by simply adding a file like `ja_JP.toml` and uncommenting `//exports.loadLanguage('xx_XX')` in langloader.js.
    - You can add only the text you wish to translate to the language file you have added. For items that are not written, the default en_US text will be used
    - This change is ready to accept Pull Requests to add toml files for various languages
- I have separated the following two files that have nothing to do with translation into separate files
    - `_placeholder.toml` for texts that do not require translation (such as placeholder news article, and default overlay description)
    - `_custom.toml` for launcher customizer (such as app title, github url, and welcome description)

**Comment by dscalzi** [comment]
Sorry this has taken a while, mostly looks good - I just need to checkout the branch and test a few things.

File: index.js [hunk 1]
```diff
@@ -236,7 +241,12 @@ function createWindow() {
     })
     remoteMain.enable(win.webContents)
 
-    ejse.data('bkid', Math.floor((Math.random() * fs.readdirSync(path.join(__dirname, 'app', 'assets', 'images', 'backgrounds')).length)))
+    const data = {
+        bkid: Math.floor((Math.random() * fs.readdirSync(path.join(__dirname, 'app', 'assets', 'images', 'backgrounds')).length)),
+        lang: (str) => ejs.render(LangLoader.queryEJS(str), data),
```
Comment (dscalzi) [inline index.js:246]: The language file should contain only strings and wildcards. I see that you're also allowing "renderable" content which really is just evaluating anything that's in the language file. This is not really secure as a malicious language file could lead to arbitrary execution.

ie logic like this
```ejs
jvmOptsLink = "https://docs.oracle.com/javase/8/docs/technotes/tools/<%= process.platform === 'win32' ? 'windows' : 'unix' %>/java.html"
```
should not be evaluated as part of the language file. Only strings with wildcards should be there.

File: app/assets/lang/_placeholder.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+overlayTitle = "Lorem Ipsum:<br>Finis Illud"
+overlayDesc = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.."
+overlayAcknowledge = "Conare Iterum"
+overlayDismiss = "Dismiss"
+
+[ejs.settings]
+versionValue = "0.0.1-alpha.18"
+memoryTotalValue = "16G"
+memoryAvailableValue = "7.3G"
+javaExecVal = "null"
```
Comment (dscalzi) [inline app/assets/lang/_placeholder.toml:19]: I think we can just leave this stuff inline

File: app/assets/lang/en_US.toml [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+serverStatusPlaceholder = "OFFLINE"
+mojangStatus = "MOJANG STATUS"
+mojangStatusTooltipTitle = "Services"
+mojangStatusNETitle = "Non&nbsp;Essential"
+newsButton = "NEWS"
+launchButton = "PLAY"
+launchButtonPlaceholder = "&#8226; No Server Selected"
+launchProgressLabel = "0%"
+launchDetails = "Please wait.."
+newsNavigationStatus = "1 of 1"
```
Comment (dscalzi) [inline app/assets/lang/en_US.toml:16]: this won't translate properly, it needs wildcards for currentPage and totalPages

**Review by dscalzi — APPROVED** [review]
LGTM, may tweak more later.

**Pull request merged**